config/config-api/src/main/java/com/thoughtworks/go/config/PluggableArtifactConfig.java

```diff
@@ -150,9 +160,8 @@ public ConfigErrors errors() {
         return errors;
     }
 
-    @Override
     public boolean hasErrors() {
-        return super.hasErrors() || !errors.isEmpty();
+        return !errors.isEmpty();
```

Comment:
Do we need `|| configuration.hasErrors()` ?

Reply:
I think so. I missed the `configuration.validateUniqueness` in `#validate`, which adds errors to the configuration object. I'll fix this.